nginx-controller/nginx/configurator.go

```diff
@@ -115,12 +117,20 @@ func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]stri
 			ProxyHideHeaders:      ingCfg.ProxyHideHeaders,
 			ProxyPassHeaders:      ingCfg.ProxyPassHeaders,
 			ServerSnippets:        ingCfg.ServerSnippets,
+			Ports:                 ports,
+			SSLPorts:              sslPorts,
 		}
 
 		if pemFile, ok := pems[serverName]; ok {
-			server.SSL = true
-			server.SSLCertificate = pemFile
-			server.SSLCertificateKey = pemFile
+			if len(sslPorts) > 1 {
```

Comment:
1. `len(sslPorts) > 1` leads to a bug, as it only enables SSL termination when we have at least 2 SSL ports.
1. this check is also unnecessary: we should always have at least 1 SSL port if SSL is enabled (`if  pemFile, ok := pems[serverName];`)

Reply:
Yes, you're right, sorry.

Reply:
That check is simple a safe guard for case when annotation specifies empty value.